slack_global_report.py

```diff
@@ -0,0 +1,241 @@
+#! /usr/bin/env python
+
+import copy
+import json
+import random
+import time
+import sys
+
+import slack
+
+import random_name
+import config
+import channel
+import channel_members_log
+import user
+import utils
+import enricher
+import slack_token
+import slack_formatter
+
+
+class SlackGlobalReport(object):
+
+    def __init__(self, fake=False):
+        self.sf = slack_formatter.SlackFormatter()
+        random.seed(time.time())
+        self.fake = fake
+        self.fake_channel = channel.Channel(fake=True)
+        self.channel = channel.Channel(fake=fake)
+        self.rn = random_name.RandomName()
+        self.user = user.User(fake=fake)
+        self.client = slack.WebClient(token=slack_token.token)
+        self.enricher = enricher.Enricher(fake=fake)
+        self.cml = channel_members_log.ChannelMembersLog()
+        self.report_channel  = self.channel.get(config.report_channel)['slack_cid']
+
+    def make_header(self, ur, pur):
+        blocks = []
+        header = "*Slack Activity Report for the Week Between {} and {}*"
+        header = header.format(ur['start_date'], ur['end_date'])
+        blocks.append(self.sf.text_block(header))
+        stats = ur['statistics']
+        posters = int(stats['posters'])
+        active_users = stats['active_users']
+        percent = (posters * 100.0) / active_users
+        text = "*{:,}/{:,}* (or *{:.1f}%*) users posted messages\n".format(posters, active_users, percent)
+        text += "Median message count was *{}*\n".format(stats['median messages'])
+        text += "The top ten posters contributed *{:.1f}%* of all messages (lower is better)\n".format(stats['topten messages'])
+        text += "The top *{}* posters (higher is better) accounted for about 50% of total volume\n".format(stats['50percent of words'])
+        blocks.append(self.sf.text_block(text))
+        w = stats['words']
+        m = stats['messages']
+        # it = interim text
+        it = "People posted *{:,}* words in *{:,}* messages (approximately *{:,}* pages), "
```

Comment:
What defines a page?

Reply:
Added a code comment -- we approximate 500 words per page

Reply:
Can you add a note somewhere in the report that explains what a page is? I know other people will ask :)

Reply:
Done